api/app/controllers/spree/api/v2/storefront/account/addresses_controller.rb

```diff
@@ -0,0 +1,71 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        module Account
+          class AddressesController < ::Spree::Api::V2::BaseController
+            before_action :require_spree_current_user
+
+            def index
+              render_serialized_payload { serialize_collection(collection) }
+            end
+
+            def create
+              result = Spree::Account::Addresses::Create.call(user: spree_current_user, address_params: address_params)
+              render_result(result)
+            end
+
+            def update
+              result = Spree::Account::Addresses::Update.call(address: resource, address_params: address_params)
+              render_result(result)
+            end
+
+            private
+
+            def collection
+              spree_current_user.addresses
+            end
+
+            def resource
+              @resource ||= collection.find(params[:id])
+            end
+
+            def collection_serializer
+              Spree::Api::Dependencies.storefront_address_serializer.constantize
+            end
+
+            def resource_serializer
+              Spree::Api::Dependencies.storefront_address_serializer.constantize
+            end
+
+            def serialize_collection(collection)
+              collection_serializer.new(collection).serializable_hash
+            end
+
+            def address_params
+              replace_country_iso_with_id(params).require(:address).permit(permitted_address_attributes)
+            end
+
+            def render_result(result)
+              if result.success?
+                render_serialized_payload { serialize_resource(result.value) }
+              else
+                render_error_payload(result.error)
+              end
+            end
+
+            def replace_country_iso_with_id(params)
```

Comment:
@kgorazd please move this part into Create / Update services - or extract this into a helper.

Also, you need to find `state_id` based on `state_name`

Reply:
fixed